src/components/AwsProvisionerHealthTable/index.jsx

```diff
@@ -0,0 +1,296 @@
+    )
+  ),
+  map(({ region, az, instanceType }) => `${region}-${az}-${instanceType}`)
+);
+
+@withStyles(theme => ({
+  drawerContainer: {
+    paddingTop: theme.spacing.double,
+    paddingBottom: theme.spacing.double,
+    width: 400,
+  },
+  headline: {
+    paddingLeft: theme.spacing.triple,
+    paddingRight: theme.spacing.triple,
+  },
+}))
+export default class AwsProvisionerHealthTable extends Component {
+  static propTypes = {
+    /** A GraphQL awsProvisionerHealth response. */
+    healthData: awsProvisionerHealth.isRequired,
+  };
+
+  state = {
+    sortBy: null,
+    sortDirection: null,
+    drawerOpen: false,
+  };
+
+  handleHeaderClick = sortBy => {
+    const toggled = this.state.sortDirection === 'desc' ? 'asc' : 'desc';
+    const sortDirection = this.state.sortBy === sortBy ? toggled : 'desc';
+
+    this.setState({ sortBy, sortDirection });
+  };
+
+  createSortedHealth = memoizeWith(
+    (healthData, sortBy, sortDirection) => {
+      const ids = sorted(healthData.running);
+
+      return `${ids.join('-')}-${sortBy}-${sortDirection}`;
+    },
+    (healthData, sortBy, sortDirection) => {
+      const healthSummary = {};
+      const { requestHealth, terminationHealth, running } = healthData;
+      const identifier = item =>
+        `${item.az}-${item.region}-${item.instanceType}`;
+      const or0 = defaultTo(0);
+
+      requestHealth.forEach(item => {
+        healthSummary[identifier(item)] = {
+          ...healthSummary[identifier(item)],
+          ...item,
+        };
+      });
+      terminationHealth.forEach(item => {
+        healthSummary[identifier(item)] = {
+          ...healthSummary[identifier(item)],
+          ...item,
+        };
+      });
+      running.forEach(item => {
+        healthSummary[identifier(item)] = {
+          ...healthSummary[identifier(item)],
+          ...item,
+        };
+      });
+
+      Object.entries(healthSummary).forEach(([key, item]) => {
+        healthSummary[key].healthy =
+          or0(item.successful) + or0(item.clean_shutdown) + or0(item.running);
+        healthSummary[key].unhealthy =
+          or0(item.failed) +
+          or0(item.spot_kill) +
+          or0(item.insufficient_capacity) +
+          or0(item.volume_limit_exceeded) +
+          or0(item.missing_ami) +
+          or0(item.startup_failed) +
+          or0(item.unknown_codes) +
+          or0(item.no_code);
+      });
+
+      const sortByProperty = camelCase(sortBy);
+      const healthValues = Object.values(healthSummary);
+
+      if (!sortBy) {
+        return healthValues;
+      }
+
+      return ifElse(
+        isEmpty,
+        identity,
+        rSort((a, b) => {
+          const firstElement =
+            sortDirection === 'desc' ? b[sortByProperty] : a[sortByProperty];
+          const secondElement =
+            sortDirection === 'desc' ? a[sortByProperty] : b[sortByProperty];
+
+          return sort(firstElement, secondElement);
+        })
+      )(healthValues);
```

Comment:
If we aren't using point-free syntax, there is little value to an `ifElse` call like this and can be more easily read imperatively.

Reply:
Agreed.